File: 0013-roman-to-integer/0013-roman-to-integer.ts
function romanToInt(s: string): number {
    const mp: Map<string, number> = new Map([
        ['I', 1],
        ['V', 5],
        ['X', 10],
        ['L', 50],
        ['C', 100],
        ['D', 500],
        ['M', 1000]
    ]);
    
    let sum = 0;
    for (let i = 0; i < s.length; i++) {
        if (mp.get(s[i])! < mp.get(s[i + 1])!) {
        sum += mp.get(s[i + 1])! - mp.get(s[i])!;
        i++;
        continue;
        }
        sum += mp.get(s[i])!;
    }
    
    return sum;
};

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        mp = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
    
        sum = 0
        i = 0
        while i < len(s):
            if i < len(s) - 1 and mp[s[i]] < mp[s[i + 1]]:
                sum += mp[s[i + 1]] - mp[s[i]]
                i += 2
            else:
                sum += mp[s[i]]
                i += 1
    
        return sum